dm/pkg/binlog/position.go

```diff
@@ -428,3 +428,7 @@ func (l *Location) SetGTID(gset gmysql.GTIDSet) error {
 func (l *Location) GetGTID() gtid.Set {
 	return l.gtidSet
 }
+
+func (l *Location) Update(gtidStr string) error {
```

Comment:
we already have `func (l *Location) SetGTID(gset gmysql.GTIDSet) error` , can we just use that func?

Reply:
syncer calls `SetGTID` when meet `query event`(ddl) or `xid event`, i.e. after the transaction is synced, there is a complete gtid set maintained in `go-mysql`.

but in validator, when meet a `gtid event`, i.e. transaction begin, we need to update the location(simpler than extending `go-mysql` to add a field in this event), so we can check whether syncer synced them

Reply:
Reasonably, update has less overhead